R/logistic_mle.R

```diff
@@ -0,0 +1,82 @@
+#'
+logistic.log_likelihood <- function(coef, x, y) {
+  n <- length(y)
+  p_x_coef <- logistic.fun(coef, x)
+  y <- matrix(y, ncol = 1)
+  sum(y * log(p_x_coef) + (1 - y) * log(1 - p_x_coef))
+}
+
+#'
+logistic.log_likelihood.gradient <- function(coef, x, y) {
+  p <- logistic.fun(coef, x)
+  t(x) %*% (y - p)
+}
+
+#'
+logistic.fun <- function(coef, x) {
```

Comment:
The `.fun` part of this name isn't that helpful, since the object itself is already a function.